### DISCUSSION TITLE
Export/import add-on

### DISCUSSION THREAD

**Discussion by remcotolsma** [pronamic/wp-pronamic-pay#123] [Ideas]
## Internal Links

- **Basecamp to-do:** https://basecamp.com/1810084/projects/18117394/todos/441951165

**Comment by remcotolsma** [comment]
Import of subscriptions was developed some time ago in https://github.com/wp-pay/core/commit/70ae3fea3d498185cfbc45ae8781eb0a645468fa, but was later removed: https://github.com/wp-pay/core/commit/509065ccde18f3fadcabb63ed6140ba1cf9fe637.

We still have to decide if we want an export/import feature and if this is something for the core library or an extra add-on plugin.

**Internal Basecamp to-do:** https://basecamp.com/1810084/projects/18117394/todos/441951706

**Comment by rvdsteege** [comment]
There's also some basic CSV importer work in https://github.com/wp-pay/importer

**Comment by rvdsteege** [comment]
From customer:

> Misschien zie ik deze functie over het hoofd, maar klopt het dat de abonnementen die staan bij de Pronamic Wordpressplugin niet geëxporteerd kunnen worden?

_Internal Help Scout ticket: https://secure.helpscout.net/conversation/1692494667/22992_

**Comment by rvdsteege** [comment]
From customer:

> Goedemorgen, is het mogelijk om een export te maken van alle actieve abonnementen? Dan kunnen we die matchen met het CRM.

_Internal Help Scout ticket: https://secure.helpscout.net/conversation/2047750802/24665/_

**Comment by rvdsteege** [comment]
From customer:

> Eerder hadden wij een export maar deze werkt inmiddels niet meer en levert een nog al onbruikbare export op.

_Internal Help Scout ticket: https://secure.helpscout.net/conversation/2091607719/24905_

**Comment by rvdsteege** [comment]
From customer:

> Graag willen wij op een eenvoudige manier alle transacties (incl. klantgegevens) kunnen exporteren echter lijkt dit niet mogelijk. Is er een optie om dit werkend te krijgen?

_Internal Help Scout ticket: https://secure.helpscout.net/conversation/2178322481/25326_

**Comment by remcotolsma** [comment]
I have tested the built-in orders export in WooCommerce, with a few orders you can download a CSV directly. With more orders WooCommerce will email the CSV once it's generated. We could make an "Analytics" menu item with an subitem "Payments" where users can set a date range. Based on `WP_Query → $found_posts` we can determine what to do (direct download or generate + email).

<img width="331" alt="Scherm­afbeelding 2023-03-09 om 20 51 31" src="https://user-images.githubusercontent.com/869674/224138546-d52b9cf5-63ed-44dc-bf41-95f5c707bc03.png">

https://github.com/woocommerce/woocommerce/blob/2896a8cd13a3aa2c7d2be2b31064607ee2d5ae8a/plugins/woocommerce-admin/client/analytics/components/report-table/index.js#L183-L239

As far as i could see the WooCommerce export feature is also REST API driven:
https://github.com/woocommerce/woocommerce/blob/2896a8cd13a3aa2c7d2be2b31064607ee2d5ae8a/packages/js/data/src/export/actions.ts#L48-L85

```
/wp-json/wc-analytics/reports/orders?orderby=net_total&order=desc&after=2022-10-01T00%3A00%3A00&before=2022-12-31T23%3A59%3A59&page=1&per_page=25&extended_info=true&_locale=user
```

```
/wp-json/wc-analytics/reports/orders/export?_locale=user
```

```json
{
    "export_id": "1678391885755"
    "message": "Your report file is being generated."
}
```

https://github.com/woocommerce/woocommerce/blob/2896a8cd13a3aa2c7d2be2b31064607ee2d5ae8a/plugins/woocommerce/src/Admin/API/Reports/Export/Controller.php#L147-L193

<img width="692" alt="Scherm­afbeelding 2023-03-09 om 21 05 07" src="https://user-images.githubusercontent.com/869674/224142075-5330b3d7-745c-4164-8c7b-aeb60e7e9dc0.png">

Link goed to:

```
/wp-admin/?action=woocommerce_admin_download_report_csv&filename=wc-orders-report-export-1678391885755
```

**Comment by rvdsteege** [comment]
From customer:

> Zouden jullie een manier willen adviseren om betalingsdata te exporteren? Het meest bruikbaar voor ons is CSV of XLS met alle velden afzonderlijk (ik denk dan aan post content die als JSON opgeslagen lijkt).Er is natuurlijk de WP export tool die XMR exporteert, maar daar hebben we weinig aan.

_Internal Help Scout ticket: https://secure.helpscout.net/conversation/2222374608/25529_

**Comment by rvdsteege** [comment]
Again, an enquiry about exporting payments:

> Bericht: Voor een relatie ben ik op zoek naar de mogelijkheid om transacties (verkoopboekingen) vanuit Pronamic te exporteren naar een `.csv` bestand (om te kunnen importeren in het boekhoudpakket). Ik zie dit in hun WordPress / Pronamic omgeving zo niet naar voren komen. Weet jij of hier een mogelijkheid voor is?

_Internal Help Scout ticket: https://secure.helpscout.net/conversation/2273005112/25771_

**Comment by kjtolsma** [comment]
Mooie functionaliteit voor een add-on?

**Comment by rvdsteege** [comment]
From https://wordpress.org/support/topic/export-transactions-csv/

> Is there a way to export all transactions in a csv file?

_Internal Help Scout ticket: https://secure.helpscout.net/conversation/2346048598/26091_

**Comment by rvdsteege** [comment]
One more customer wanting to export payments:

> Echter zouden we graag het tabblad van 'betalingen' kunnen exporteren voor onze boekhouder.
Is er een mogelijkheid dat deze functie ingebouwd word?

_Internal Help Scout ticket: https://secure.helpscout.net/conversation/2477584719/26743_

**Comment by remcotolsma** [comment]
For Pronamic Pay Premium users we have a "Pronamic Pay WP All Export Add-On" plugin available. If users want to export very specific payment and/or subscription data, we can add this if necessary.
